src/components/ModelCollapse.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Original file: https://github.com/swagger-api/swagger-ui/blob/3.x/src/core/components/model-collapse.jsx
+ * @prettier
+ */
+
+import React, { Component } from "react"
+
+export default class ModelCollapse extends Component {
+  constructor(props, context) {
+    super(props, context)
+
+    let { expanded, collapsedContent } = this.props
+
+    this.state = {
+      expanded: expanded,
+      collapsedContent: collapsedContent || ModelCollapse.defaultProps.collapsedContent
+    }
+  }
+
+  componentDidMount() {
+    const { hideSelfOnExpand, expanded, modelName } = this.props
+    if (hideSelfOnExpand && expanded) {
+      // We just mounted pre-expanded, and we won't be going back..
+      // So let's give our parent an `onToggle` call..
+      // Since otherwise it will never be called.
+      this.props.onToggle(modelName, expanded)
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.expanded !== nextProps.expanded) {
+      this.setState({ expanded: nextProps.expanded })
+    }
+  }
+
+  handleKeyDownToggle = (event) => {
+    if (event.nativeEvent.code === "Enter" || event.nativeEvent.code === "Space") {
+      this.toggleCollapsed()
+    }
+  }
+
+  toggleCollapsed = () => {
+    if (this.props.onToggle) {
+      this.props.onToggle(this.props.modelName, !this.state.expanded)
+    }
+
+    this.setState({
+      expanded: !this.state.expanded
+    })
+  }
+
+  render() {
+    const { title, classes } = this.props
+
+
+    return (
+      <div className={classes || ""}>
+        {this.state.expanded && this.props.hideSelfOnExpand ? this.props.children : (
+          <div>
+            {title &&
+              <div
+                role="button"
```

Comment:
Any reason we didn't use a `button`?

Reply:
@codelemur mainly to reduce the chance of there being any styling issues that arise from changing from a div to a button

Reply:
But previously (In original ModelCollapse) it was a `button` so why it might cause any styling issue?

Reply:
@tomaszwylezek I think @davidma415 is thinking that between the swaggerUI CSS and kongponents CSS, somewhere along the line someone might affect the styles of these buttons unintentionally.

I lean towards semantic HTML, but our project has a lot of side effects that can make things weird, so I was gonna let this one go :P

Reply:
Okay I think I might have linked the wrong files lol, I linked the actual swagger-ui repo, but I meant to link our fork. 

https://github.com/Kong/swagger-ui/blob/main/src/core/components/model-collapse.jsx

Reply:
But yeah @tomaszwylezek @codelemur I just didn't want to affect the code unintentionally. And in the _correct_ original file, it was just a span, so the simplest was to change it to an inline div.